### PR TITLE
User Profile Setting Menu

### PR DESCRIPTION
**Pull request by akoushke**
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-61527

### CODE REVIEW

File: packages/node_modules/@ciscospark/widget-recents/src/container.js [hunk 1]
```diff
@@ -77,8 +77,8 @@ export const ownPropTypes = {
 const defaultProps = {
   extendedLoad: true,
   enableAddButton: false,
-  enableUserProfile: true,
-  enableSpaceListFilter: true,
+  enableUserProfile: false,
+  enableSpaceListFilter: false,
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/container.js]: Why are we changing the default props? We've released these already and this breaks the existing usage.

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js [hunk 2]
```diff
@@ -2,31 +2,34 @@ import React from 'react';
...
 import classNames from 'classnames';
 
-import {Button, Icon, SearchInput} from '@collab-ui/react';
+import {Button, Icon, Popover, SearchInput} from '@collab-ui/react';
 
 import PresenceAvatar from '@ciscospark/react-container-presence-avatar';
 
-
 import styles from './RecentsHeader.css';
+import PopoverContent from './PopoverContent.js';
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: You don't need the `.js` here. Also, can you move it above the styles import? I don't think there's an actual "rule" but the style import is the last import on (hopefully all) of our components.

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js [hunk 3]
```diff
@@ -38,20 +41,29 @@ function RecentsHeader(props) {
...
   } = props;
 
+  const content = <PopoverContent currentUser={currentUser} onSignOutClick={onSignOutClick} />;
+
   return (
     <div className={classNames('ciscospark-recents-header', styles.recentsHeader, 'cui--dark', hideBottomBorder ? '' : styles.bottomBorder)}>
       <div className={classNames('ciscospark-recents-header-left', styles.headerSideItem)}>
-        {
-          enableUserProfile &&
+        { enableUserProfile &&
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: We need a config for enabling the popover. Just having the user profile enabled shouldn't show the popover.

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css [hunk 4]
```diff
@@ -39,3 +39,8 @@
   min-width: 40px;
   height: 40px;
 }
+
+/* To remove default native browser blue outline on button focus state*/
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css]: This was requested by the collab-ui team. We pressed back on it, but they insisted, so it needs to stay.

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css [hunk 5]
```diff
@@ -1,14 +1,14 @@
 .recentsHeader {
-  height: 60px;
+  height: 70px;
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css]: Web client header is 60px. Why the change?

File: packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.test.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+describe('PopoverContent component', () => {
+  beforeEach(() => {
+    props = {
+      currentUser: {
+        id: 'my-id',
+        displayName: 'Snapshot User',
+        img: 'img'
+      },
+      enableAddButton: true,
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.test.js]: Popover content doesn't have any of these props. `onSignoutClick` is missing too.

File: packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+
+function PopoverContent(props) {
+  const {
+    currentUser,
+    onSignOutClick
+  } = props;
+
+  /* eslint-disable jsx-a11y/click-events-have-key-events */
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.js]: Why are we disabling accessibility features?

File: packages/node_modules/@ciscospark/widget-recents/src/container.js (hunk 1 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/container.js]: screen shot 2019-03-07 at 9 32 13 am
The default props are false. I understand that this was merged before. But it contradicts with what we have in mind for design.

Comment (adamweeks) [reply packages/node_modules/@ciscospark/widget-recents/src/container.js]: Default props for the widget and default props for the demo are not the same. Check out the documented defaults: https://github.com/webex/react-ciscospark/tree/master/packages/node_modules/@ciscospark/widget-recents#configuration

Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/container.js]: so what are default props for widget? if thats the case then we need to stay consistent with the widget demo. As far as I remember, we decided not to show any header unless the user wanted to

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css (hunk 4 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css]: according to the mockups and web client, there is no blue outline on focus state.

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css (hunk 5 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.css]: yes. 60px is fine.

File: packages/node_modules/@ciscospark/widget-recents/src/container.js (hunk 1 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/container.js]: addressed

File: packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.js (hunk 7 above)
Comment (adamweeks) [reply packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.js]: Signout div still needs keyboard handling

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js [hunk 8]
```diff
@@ -2,56 +2,110 @@ import React from 'react';
...
+ * Displays the popover content if needed
+ *
+ * @param {bool} enableUserProfile to display the profile avatar
+ * @param {bool} enableUserProfilePopover to display the profile popover content
+ * @param {func} onProfileClick to notify if the avtar had been clicked
+ * @param {func} onSignOutClick to notify if the sign out link had been clicked
+ * @param {obj} currentUser current user information
+ * @returns {obj} JSX element with or without avatar profile
+ */
+function displayProfileAvatar(
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: Can you use [property destructuring](https://github.com/webex/web-styleguide/tree/master/javascript#destructuring--object) here? Not sure why eslint didn't catch this one.

File: packages/node_modules/@ciscospark/widget-recents/src/enhancers/errors.js [hunk 9]
```diff
@@ -61,6 +61,24 @@ function checkForErrors(props) {
   }
 }
 
+function checkForPropsErrors(props) {
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/enhancers/errors.js]: Nice!

File: packages/node_modules/@ciscospark/widget-recents/src/selector.js [hunk 10]
```diff
@@ -211,6 +211,18 @@ const getRecentSpacesWithDetail = createSelector(
   }
 );
 
+const getCurrentUserWithAvatar = createSelector(
+  [getCurrentUser, getAvatars], (currentUser, avatars) => {
+    let user;
+
+    if (currentUser && currentUser.id) {
+      user = Object.assign(currentUser, {img: avatars.getIn(['items', currentUser.id])});
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/selector.js]: This is doing the same thing as adding the properties to the object, you want to do `Object.assign({}, currentUser, {img` so that the "[target](http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)" is a new js object.

File: test/journeys/specs/smoke/widget-recents/index.js [hunk 11]
```diff
@@ -308,6 +308,18 @@ describe('Smoke Tests - Recents Widget', () => {
         assert.isNotEmpty(event.email, 'profile:clicked data does not contain email');
         assert.isNotEmpty(event.orgId, 'profile:clicked data does not contain orgId');
       });
+
+      it('user_signout:clicked', () => {
```
Comment (adamweeks) [inline test/journeys/specs/smoke/widget-recents/index.js]: I think your test failure is because you didn't init the widget with the profile popover enabled. Check line 64

File: packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.js (hunk 7 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/PopoverContent.js]: we didn't talk about adding a keyboard handling to signout. and Other clients don't even support that. I don't see a point adding a keyboard handling to signout. Unless its just a an empty function.

**Comment by akoushke** [comment]
> Looks like tests are failing. Could you also add to the [startup settings tests](https://github.com/webex/react-ciscospark/blob/master/test/journeys/specs/recents/global/startup-settings.js#L39)?

I don't know how to enable the popover content on a startup, since we need to click on profile avatar.

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js (hunk 8 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: property destruction occurs here:
```js
const {
    currentUser,
    enableAddButton,
    enableSpaceListFilter,
    enableUserProfile,
    enableUserProfilePopover,
    hideBottomBorder,
    onAddClick,
    onFilterChange,
    onProfileClick,
    onSignOutClick
  } = props;
```
I could not send the props to our helper method, eslint complains that the props have been passed to helper method are being declared but not used.

Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: Therefore, I destruct the property first and pass the items one by one:
```js
 {displayProfileAvatar(
          enableUserProfile,
          enableUserProfilePopover,
          onProfileClick,
          onSignOutClick,
          currentUser
        )}
```

File: packages/node_modules/@ciscospark/widget-recents/src/selector.js (hunk 10 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/selector.js]: I don't think thats what it does. it merges the new object `{img...` with `currentUser` and returns a new object which is what I'm passing.

Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/selector.js]: Adam, I cant make it work on my machine. your suggested statement does not reflect the currentUser properties into a new object. you are more than welcome to make the changes.

Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/selector.js]: I'm keeping it as is.

File: test/journeys/specs/smoke/widget-recents/index.js (hunk 11 above)
Comment (akoushke) [reply test/journeys/specs/smoke/widget-recents/index.js]: I'm trying.

**Comment by akoushke** [comment]
I'm just sad....

**Comment by adamweeks** [comment]
The keyboard handling is to pass the accessibility checks, not something by
design. That is why the eslint is throwing an error that you’ve disabled.

File: test/journeys/lib/test-helpers/recents-widget/index.js [hunk 12]
```diff
@@ -18,6 +18,7 @@ export const elements = {
   clearButton: 'button[aria-label="clear input"]',
   headerBar: '.ciscospark-recents-header',
   headerProfile: '.ciscospark-recents-header .ciscospark-recents-header-left button',
+  headerSignout: 'ciscospark-recents-header-profile-popover-sign-out',
```
Comment (akoushke) [inline test/journeys/lib/test-helpers/recents-widget/index.js]: @adamweeks  this is my element.

File: test/journeys/specs/smoke/widget-recents/index.js [hunk 13]
```diff
@@ -308,6 +309,18 @@ describe('Smoke Tests - Recents Widget', () => {
...
+        clearEventLog(browserLocal);
+        browserLocal.element(elements.headerProfile).click();
```
Comment (akoushke) [inline test/journeys/specs/smoke/widget-recents/index.js]: click the headerprofile
then click the signout

File: test/journeys/specs/smoke/widget-recents/index.js (hunk 11 above)
Comment (akoushke) [reply test/journeys/specs/smoke/widget-recents/index.js]: it worked.

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js (hunk 8 above)
Comment (adamweeks) [reply packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: I meant the property destructuring in the props of the function like in the example:

```js
// best
function getFullName({ firstName, lastName }) {
  return `${firstName} ${lastName}`;
}
```

File: packages/node_modules/@ciscospark/widget-recents/src/selector.js (hunk 10 above)
Comment (adamweeks) [reply packages/node_modules/@ciscospark/widget-recents/src/selector.js]: Just tried it out locally, that is because the current user object is a immutable record. This works as expected:

```js
user = Object.assign({}, currentUser.toJS(), {img: avatars.getIn(['items', currentUser.id])});
```

**Comment by akoushke** [comment]
as I mentioned in the past `toJS()` does not work in my machine. I can change it the way you want but Im not gonna be able to test it in my machine. I will push it based on your claim.

File: packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js [hunk 14]
```diff
@@ -58,11 +58,12 @@ class DemoWidget extends Component {
       composerActions,
       destinationId,
       initialActivity: isMeetOnly ? 'meet' : cookies.get('initialActivity') || 'message',
-      enableSpaceListFilter: cookies.get('enableSpaceListFilter') === 'true',
+      enableSpaceListFilter: cookies.get('enableSpaceListFilter') === 'true' || true,
```
[the remark this answers is not in the document]
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js]: break what exactly? this is what we have in default in our README. the demo page should not contradict with what we have in README. it didn't break anything for me.

[the remark this answers is not in the document]
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js]: also I remember asking you and you told me feel free to change them... and I did.

[the remark this answers is not in the document]
Comment (adamweeks) [reply packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js]: If the user sets the config manually to false, when they return to the demo page, it will read from the cookie and see that it is 'false", but it will not actually set the state value to false because of the added `|| true`.

**Comment by akoushke** [comment]
> @adamweeks as I mentioned to you earlier `toJS()` does not work in my machine. I can change it the way you want but Im not gonna be able to test it in my machine. I will push it based on your claim. I showed you the error that I was getting before and you have no solution, therefore, i kept it as is:
> 

Also did you pull from my branch and `npm i` and run it?

**Comment by adamweeks** [comment]
> @adamweeks as I mentioned to you earlier `toJS()` does not work in my machine. I can change it the way you want but Im not gonna be able to test it in my machine. I will push it based on your claim. I showed you the error that I was getting before and you have no solution, therefore, i kept it as is:
> 

@akoushke I understand that it isn't working on your machine. Let's figure out why. It is the proper way to convert an immutable object to a javascript object. I checked out your branch and added the above code to get it working. 

Maybe we have a version mismatch? I have v3.8.2 installed

**Comment by akoushke** [comment]
node version? @adamweeks

**Comment by akoushke** [comment]
I'm on this:
Screen Shot 2019-03-08 at 2 49 34 PM

**Comment by akoushke** [comment]
> > @adamweeks as I mentioned to you earlier `toJS()` does not work in my machine. I can change it the way you want but Im not gonna be able to test it in my machine. I will push it based on your claim. I showed you the error that I was getting before and you have no solution, therefore, i kept it as is:
> > 
> 
> @akoushke I understand that it isn't working on your machine. Let's figure out why. It is the proper way to convert an immutable object to a javascript object. I checked out your branch and added the above code to get it working.
> 
> Maybe we have a version mismatch? I have v3.8.2 installed

immutable you mean. duhh. let me see

File: packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js [hunk 15]
```diff
@@ -58,11 +58,12 @@ class DemoWidget extends Component {
...
       extendedLoad: cookies.get('extendedLoad') === 'true',
       mode,
       recentsEnableAddButton: cookies.get('recentsEnableAddButton') === 'true',
-      recentsEnableUserProfile: cookies.get('recentsEnableUserProfile') === 'true',
+      recentsEnableUserProfile: cookies.get('recentsEnableUserProfile') === 'true' || true,
+      recentsEnableUserProfilePopover: cookies.get('recentsEnableUserProfilePopover') === 'true',
```
[the remark this answers is not in the document]
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js]: it does not apply here.

**Comment by akoushke** [comment]
> > > @adamweeks as I mentioned to you earlier `toJS()` does not work in my machine. I can change it the way you want but Im not gonna be able to test it in my machine. I will push it based on your claim. I showed you the error that I was getting before and you have no solution, therefore, i kept it as is:
> > > 
> > 
> > 
> > @akoushke I understand that it isn't working on your machine. Let's figure out why. It is the proper way to convert an immutable object to a javascript object. I checked out your branch and added the above code to get it working.
> > Maybe we have a version mismatch? I have v3.8.2 installed
> 
> immutable you mean. duhh. let me see
> 
> Screen Shot 2019-03-08 at 2 57 24 PM

the App also crashes on Netlify using `toJS()`
Screen Shot 2019-03-10 at 7 13 35 PM

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js (hunk 8 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: not related anymore

File: packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js]: I will update the code to not break the functionality.

File: packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js [hunk 16]
```diff
@@ -58,11 +58,12 @@ class DemoWidget extends Component {
       composerActions,
       destinationId,
       initialActivity: isMeetOnly ? 'meet' : cookies.get('initialActivity') || 'message',
-      enableSpaceListFilter: cookies.get('enableSpaceListFilter') === 'true',
+      enableSpaceListFilter: cookies.get('enableSpaceListFilter') === 'true' || true,
       extendedLoad: cookies.get('extendedLoad') === 'true',
       mode,
       recentsEnableAddButton: cookies.get('recentsEnableAddButton') === 'true',
-      recentsEnableUserProfile: cookies.get('recentsEnableUserProfile') === 'true',
+      recentsEnableUserProfile: cookies.get('recentsEnableUserProfile') === 'true' || true,
```
[the remark this answers is not in the document]
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-demo/src/components/demo-widget/index.js]: I will update the code to not break the functionality

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js [hunk 17]
```diff
@@ -4,54 +4,47 @@ import classNames from 'classnames';
...
       <div className={classNames('ciscospark-recents-header-left', styles.headerSideItem)}>
-        {
-          enableUserProfile &&
-          <PresenceAvatar
-            avatarId={currentUser.id}
-            name={currentUser.displayName}
-            onClick={onProfileClick}
-          />
-        }
+        {<UserProfileAvatar {...props} /> }
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: Can we specifically pass the props here? We typically only pass the props object when we have react components that are meant to be more of a higher-order component type wrapper component. This helps us know if we have prop naming issues in the future with changes.

File: packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+        <MenuItem
+          className={classNames('ciscospark-recents-header-profile-menu-signout', styles.signout)}
+          onClick={onSignOutClick}
+          label="Sign Out"
+        />
+      </Menu>
+    </MenuOverlay>
+  );
+
+  const profileAvatar = enableUserProfile ? presenceAvatar : null;
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js]: Is this check necessary? We throw an error if `enableUserProfileMenu` is enabled but `enableUserProfile` isn't.

My suggestion would be to leave the if check in `RecentsHeader.js` for the avatar, and only check for the menu prop here.

```js
      {
          enableUserProfile &&
          <UserProfileAvatar
```

File: packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+  const profileAvatarWithMenu = enableUserProfileMenu ? menu : profileAvatar;
+
+  return profileAvatarWithMenu;
+}
+
+UserProfileAvatar.propTypes = propTypes;
+UserProfileAvatar.defaultProps = defaultProps;
+
+export default UserProfileAvatar;
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js]: I like pulling this out of the RecentsHeader code!

File: packages/node_modules/@ciscospark/widget-recents/src/components/__snapshots__/UserProfileAvatar.test.js.snap [hunk 20]
```diff
@@ -0,0 +1,161 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`ProfileAvatar component displays an error page if userProfile is disabled and userProfileMenu is enabled 1`] = `
+<MenuOverlay
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/components/__snapshots__/UserProfileAvatar.test.js.snap]: This isn't an error page

File: packages/node_modules/@ciscospark/widget-recents/src/selector.js [hunk 21]
```diff
@@ -211,6 +211,19 @@ const getRecentSpacesWithDetail = createSelector(
   }
 );
 
+const getCurrentUserWithAvatar = createSelector(
+  [getCurrentUser, getAvatars], (currentUser, avatars) => {
+    let user;
+
+    if (currentUser && currentUser.id) {
+      // user = Object.assign(currentUser, {img: avatars.getIn(['items', currentUser.id])});
+      user = Object.assign({}, currentUser.toJS(), {img: avatars.getIn(['items', currentUser.id])});
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/selector.js]: Is this working now?

File: packages/node_modules/@ciscospark/widget-recents/src/events.js [hunk 22]
```diff
@@ -10,7 +10,8 @@ export const eventNames = {
   MEMBERSHIPS_CREATED: 'memberships:created',
   MEMBERSHIPS_DELETED: 'memberships:deleted',
   ADD_CLICKED: 'add:clicked',
-  PROFILE_CLICKED: 'profile:clicked'
+  PROFILE_CLICKED: 'profile:clicked',
+  USER_SIGNOUT_CLICKED: 'user_signout:clicked'
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/events.js]: Don't forget to update the EVENTS.md file with this new event

File: packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js (hunk 17 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/RecentsHeader.js]: yeah I can do that.

File: packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js (hunk 18 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js]: discussing about it in teams.

File: packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js (hunk 19 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js]: I'm not following...

File: packages/node_modules/@ciscospark/widget-recents/src/components/__snapshots__/UserProfileAvatar.test.js.snap (hunk 20 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/components/__snapshots__/UserProfileAvatar.test.js.snap]: right. we should not worry about this case.

File: packages/node_modules/@ciscospark/widget-recents/src/selector.js (hunk 21 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/selector.js]: no. it does not also work on Netlify

File: packages/node_modules/@ciscospark/widget-recents/src/events.js (hunk 22 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/events.js]: yep. Just did.

File: packages/node_modules/@ciscospark/widget-recents/src/selector.js (hunk 21 above)
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/selector.js]: I'm curious, on how you made it work on your machine. did you open the recent widget? thats when it crashes.

File: packages/node_modules/@ciscospark/widget-recents/src/container.js [hunk 23]
```diff
@@ -171,12 +174,16 @@ export class RecentsWidget extends Component {
...
+      currentUserWithAvatar
     } = this.props;
 
-    this.handleEvent(eventNames.PROFILE_CLICKED, currentUser.toJS());
+    this.handleEvent(eventNames.PROFILE_CLICKED, currentUserWithAvatar);
   }
 
+  @autobind
+  handleSignOutClick() {
+    this.handleEvent(eventNames.USER_SIGNOUT_CLICKED, {});
```
[the remark this answers is not in the document]
Comment (akoushke) [reply packages/node_modules/@ciscospark/widget-recents/src/container.js]: addressed in teams

File: packages/node_modules/@ciscospark/react-component-presence-avatar/src/index.js [hunk 24]
```diff
@@ -64,7 +64,7 @@ function PresenceAvatar(props) {
         type = 'dnd';
         break;
       default:
-        type = 'default';
+        type = '';
```
[the remark this answers is not in the document]
Comment (akoushke) [reply packages/node_modules/@ciscospark/react-component-presence-avatar/src/index.js]: yeah I created a ticket: https://github.com/collab-ui/collab-ui/issues/123

**Comment by adamweeks** [comment]
@akoushke can't tell if you already did, but you'll have to rebase this on master since Jenkinsfile was updated in another PR.
